src/mlpack/core/util/size_checks.hpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file size_checks.hpp
+ * @author Kirill Mishchenko
+ * @author Bisakh Mondal
+ *
+ * Utility for checking same size & same dimensionality.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_UTIL_SIZE_CHECKS_HPP
+#define MLPACK_UTIL_SIZE_CHECKS_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace util {
+
+/**
+  * Check for if the given data points & labels have same size.
+  *
+  * @param data data.
+  * @param labels Labels.
+  * @param callerDescription A description of the caller that can be used for
+  *     error generation.
+  * @param addInfo  Additional information about labels that can be used for
+  *	precise error generation. Default is "labels". Another e.g. weights
+  */
```

Comment:
```suggestion
 * Check for if the given data points & labels have same size.
 *
 * @param data data.
 * @param labels Labels.
 * @param callerDescription A description of the caller that can be used for
 *     error generation.
 * @param addInfo Name to use for labels for precise error generation. Default
 *     is "labels"; for example, "weights" could also be used.
 */
```
Just some more simple spacing and grammar fixes. :)